### ISSUE TITLE
Support for bridge-networks with netmask smaller than 16 bits 

### ISSUE THREAD

**Issue by marvaini** [vmware/vic#3943]
#For stories, please include the information below:

**User Statement:**

We would like to provide more concurrent users with separate vCH environments with less containers. To be able to do so without complex network gear integration or NSX we would like to be able to create bridge-networks with smaller netmasks (value higher than 16 bits).

**Details:**

**Acceptance Criteria:**
Support for bridge-network range with subnet mask value higher than 16 bits.

#For bug reports, please include the information below:

**VIC version:**

0.8.0

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?

**Steps to reproduce:**

**Actual behavior:**

**Expected behavior:**

**Logs:**

**Additional details as necessary:**

**Comment by hickeng** [comment]
@hmahmood Can you give me a quick status/info dump for this issue.
I tried creating a VCH with ` --bridge-network-range=172.16.0.0/16 ` which succeeded, however the default bridge network is created using the entire range:
```
[
    {
        "Name": "bridge",
        "Id": "45ceb9714749aa4870958127b6cc525b9b4125f0b263e5218348a20c0594cf68",
        "Created": "2017-09-21T03:01:33.460614012Z",
        "Scope": "",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.16.0.0/16",
                    "Gateway": "172.16.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]
```

and when I try running `docker network create net1` we get:
```
Error response from daemon: could not find IP range for mask ffff0000
```

Given the details of the issue I was expecting to be able to use at least a /16 range and sub-allocate out of that for the various bridge networks. I did look for and didn't find an option to adjust the width of the default bridge network that gets created.

**Comment by hmahmood** [comment]
The default width for bridge networks is `/16`, which is why you are seeing only one network created above. We have #3771 tracking the addition of an option to vic-machine to allow for smaller widths.

**Comment by malikkal** [comment]
Would this be fixed with 1.4.x, please? Thank you.

**Comment by hickeng** [comment]
I've marked this as a p2 (tentative) simply because of the date associated with the release, but we really should aim to include this in 1.4.1 if possible.
This is both an ease-of-use item (allowing more reuse of assigned VLANs) and to avoid resource exhaustion (VLANs are a limited resource and may be non-trivial to request).

**Comment by malikkal** [comment]
I am really in need of this.

any luck if this can be cherry picked into 1.4.2, please? :pray:

Thank you.

**Comment by renmaosheng** [comment]
Hi Malikkal,
    1.4.2 turns out to be an emergent patch which we fixes a harbor catastrophic issue which will be released soon.  we will consider this to 1.4.3 normal patch

**Comment by yuyangbj** [comment]
@malikkal I have submitted a PR to support it! Currently, I added a parameter '--bridge-network-width' to work with --bridge-network-range, so you can use these two parameters to change the default bridge network range. Is it what you want? Appreciate your response.

**Comment by malikkal** [comment]
@yuyangbj if this helps create VCH with bridge networks smaller than /12 (for example /16) and still docker network create works; then yes, this is what is required. If yes, is there a build that I could test with, please? OR would this be included in 1.5.x? Thank you.
